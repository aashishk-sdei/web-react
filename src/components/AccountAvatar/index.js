import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Avatar from "@material-ui/core/Avatar";
import { grey } from "@material-ui/core/colors";

const colors = ["#5199B5", "#388367", "#885F90", "#B88B82", "#8A4831", "#BC8A40"];
const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
    width: 32,
    height: 32,
    "& > *": {
      margin: 0,
    },
  },
  small: {
    width: theme.spacing(4),
    height: theme.spacing(4),
  },
  grey: (props) => {
    return {
      color: theme.palette.getContrastText(grey[900]),
      backgroundColor: props.bgColorCode || props.bgColorCode === 0 ? colors[props.bgColorCode] : grey[900],
      width: theme.spacing(4),
      height: theme.spacing(4),
      fontSize: props.likedPersons && "12px",
    };
  },
}));

const AccountAvatar = ({ avatarName, imgSrc, ...props }) => {
  const classes = useStyles(props);

  return <div className={props.avatarsizeclass?props.avatarsizeclass:"avtar-circle-medium"}>{imgSrc ? <Avatar src={imgSrc} className={classes.small} /> : <Avatar className={classes.grey}>{avatarName.toUpperCase()}</Avatar>}</div>;
};

export default AccountAvatar;
