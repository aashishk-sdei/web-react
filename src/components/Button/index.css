/* Common */
.btn {
  @apply font-medium py-1.5 px-4 rounded-lg border border-transparent focus:outline-blue  flex items-center;
}

/* Button Type */
.btn-default {
  @apply bg-gray-1 text-gray-7 hover:bg-gray-2;
}

.btn-default-dark {
  @apply bg-gray-2 text-blue-4 hover:bg-gray-3;
}

.btn-primary-inverted {
  @apply bg-gray-1 text-blue-4 hover:bg-gray-2;
}

.btn-primary {
  @apply bg-blue-4 text-white hover:bg-blue-5;
}

.btn-secondary {
  @apply bg-gray-7 text-white hover:bg-black;
}

.btn-red {
  @apply bg-redShades-4 border-transparent text-white hover:bg-redShades-5;
}
.btn-red-dark {
  @apply bg-redShades-6 border-transparent text-white hover:bg-redShades-7;
}

.btn-danger {
  @apply bg-maroon-5 text-white hover:bg-maroon-6;
}
.btn-success {
  @apply bg-green-4 text-white hover:bg-green-6;
}
.btn-skyblue {
  @apply bg-skyblue-6 border-skyblue-6 text-white hover:bg-skyblue-7 hover:border-skyblue-7;
}
.btn-white {
  @apply bg-white border-white text-redShades-7 hover:bg-gray-2 hover:border-gray-2 hover:text-redShades-7;
}
.link-white {
  @apply px-3 bg-transparent border-transparent text-white hover:bg-transparent hover:border-transparent hover:text-gray-2;
}
.link-white .icon-large {
  @apply mt-auto mb-auto;
}
.link-white .icon-large svg {
  @apply w-4;
}
.link-white .swapper {
  @apply flex-row-reverse;
}
.link-white .ml-swapper {
  @apply ml-0 mr-2;
}
.link-danger{
  @apply bg-gray-1 text-maroon-5 hover:bg-gray-2 hover:text-maroon-6;
}
.btn-maroon {
  @apply bg-maroon-4;
}

.btn-stepper {
  @apply bg-gray-1 border border-gray-3 py-1.5 px-4 rounded-lg focus:outline-none focus:ring-2 focus:ring-blue-5 flex items-center text-gray-4 hover:bg-blue-1 justify-center hover:text-blue-4 h-20 !important w-80;
}

/* Button Size */
.btn-small {
  @apply text-xs h-6;
}

.btn-medium {
  @apply text-sm h-8;
}

.btn-large {
  @apply text-base h-10;
}
.btn-xlarge {
  @apply text-2xl h-16 md:h-21.5 px-6;
}

/* Button Disable */
.btn-disable {
  @apply opacity-50 cursor-default;
  filter: opacity(0.8) !important;
}
.link-gray{
  @apply text-gray-7 cursor-pointer no-underline hover:text-blue-4 hover:underline;
}