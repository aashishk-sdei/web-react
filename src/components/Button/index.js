import React, { useState } from "react";
import PropTypes from "prop-types";
import "./index.css";

// Components
import Loader from "../Loader";
import Icon from "../Icon";
import Translator from "../Translator";

const Button = ({ type, title, size, width, icon, disabled, handleClick, loading, tkey, buttonType, _textColor, fontWeight, ...props }) => {
  const getBtnType = () => {
    switch (type) {
      case "default":
        return "btn-default";
      case "default-dark":
        return "btn-default-dark";
      case "primary":
        return "btn-primary";
      case "primary-inverted":
        return "btn-primary-inverted";
      case "secondary":
        return "btn-secondary";
      case "danger":
        return "btn-danger";
        case "success":
          return "btn-success";
      case "red":
        return "btn-red";
      case "red-dark":
        return "btn-red-dark";
      case "skyblue":
        return "btn-skyblue";
      case "white":
        return "btn-white";
      case "link-white":
        return "link-white";
      case "link-danger":
        return "link-danger";
      case "maroon":
        return "btn-maroon";
      case "stepper":
        return "btn-stepper";
      default:
        return "";
    }
  };
  const btnType = getBtnType();

  const getBtnSize = () => {
    switch (size) {
      case "small":
        return "btn-small";
      case "medium":
        return "btn-medium";
      case "large":
        return "btn-large";
      case "xlarge":
        return "btn-xlarge typo-font-bold";
      default:
        return "";
    }
  };
  const btnSize = getBtnSize();

  const getBtnWidth = () => {
    return (width === "full")?"w-full":""
  };
  const btnWidth = getBtnWidth();

  const btnDisable = disabled || loading ? "btn-disable" : "";

  const getlocalIconColor = () => {
    switch (type) {
      case "default":
        return "secondary";
      case "stepper":
        return "default";
      default:
        return "white";
    }
  };
  const localIconColor = getlocalIconColor();

  const [iconColor, setIconColor] = useState(localIconColor);

  const handleMouseEnter = () => {
    if (type === "stepper") setIconColor("primary");
  };

  const handleMouseLeave = () => {
    if (type === "stepper") setIconColor(localIconColor);
  };

  const getTitle = (tkeyin) => {
    return tkeyin !== "" ? <Translator tkey={tkeyin} /> : title;
  };

  const getFontWeight = () => {
    switch (fontWeight) {
      case "regular":
        return "typo-font-regular";
      case "light":
        return "typo-font-light";
      case "medium":
        return "typo-font-medium";
      case "bold":
        return "typo-font-bold";
      case "lyon-regular":
        return "lyon-font-regular";
      case "lyon-medium":
        return "lyon-font-medium";
      default:
        return "typo-font-medium";
    }
  };
  return (
      <button type={buttonType ? buttonType : "button"} className={`btn ${btnType} ${btnSize} ${btnWidth} ${btnDisable}`} onClick={!disabled ? handleClick : undefined} onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave} {...props}>
      {icon ? (
        <div className="flex items-center swapper">
          <Icon type={icon} color={iconColor} size={size} />
          <div className={`ml-2 typo-font-light ml-swapper ${getFontWeight()}`} > {getTitle(tkey)}</div>
        </div>
      ) : (
        <div className="flex justify-center items-center w-full">
          {loading && (
            <span className="mr-2">
              <Loader spinner={true} color="default" size={12} />
            </span>
          )}
          <span className={getFontWeight()}>{getTitle(tkey)}</span>
        </div>
      )}
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.oneOf(["default", "default-dark", "primary", "primary-inverted", "secondary", "danger", "success", "skyblue", "white", "link-white","link-danger", "red", "red-dark", "stepper"]),
  title: PropTypes.string,
  size: PropTypes.oneOf(["small", "medium", "large", "xlarge"]),
  icon: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  disabled: PropTypes.bool,
  handleClick: PropTypes.func,
  tkey: PropTypes.string,
};

Button.defaultProps = {
  type: "primary",
  title: "Learn more",
  size: "medium",
  icon: false,
  disabled: false,
  handleClick: undefined,
  tkey: "",
};

export default Button;
