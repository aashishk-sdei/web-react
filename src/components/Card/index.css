.wa-focused-card {
  @apply max-w-md p-2 bg-gray-7 rounded-md overflow-hidden shadow-focus-card hover:shadow-inner inline-block;
}
.focus-pointer{
  @apply cursor-pointer
}
.living {
  max-height: -webkit-fill-available;
  max-width: -webkit-fill-available;
}
.deceased {
  max-height: -webkit-fill-available;
  max-width: -webkit-fill-available;
}
.wa-focused-card-menu {
  @apply -ml-2 -mt-2 max-w-md p-2 bg-gray-7 rounded-md overflow-hidden inline-block cursor-pointer shadow-focus-card;
}

.wa-card-medium {
  @apply bg-white h-12 max-h-fitcontent ml-0.5 mr-0.5 rounded-md overflow-hidden flex items-center shadow-card hover:shadow-inner cursor-pointer;
}
.wa-card-medium-menu {
  @apply bg-white h-12 max-h-fitcontent ml-0.5 mr-0.5 rounded-md overflow-hidden flex items-center shadow-card;
}

.wa-card-small {
  @apply bg-white max-h-fitcontent rounded-md ml-0.5 mr-0.5 overflow-hidden flex items-center shadow-card hover:shadow-inner cursor-pointer;
}

.wa-card-small-menu {
  @apply bg-white max-h-fitcontent rounded-md ml-0.5 mr-0.5 overflow-hidden flex items-center shadow-card;
}

.wa-card-nextGen {
  @apply bg-gray-2 h-11.75 max-h-fitcontent rounded-md overflow-hidden items-center border-dashed border-2 border-gray-4 text-gray-5;
}

.wa-card-nextGen-hover {
  @apply bg-gray-2  max-h-fitcontent rounded-md overflow-hidden items-center border-dashed border-2 border-gray-4 text-gray-5 hover:bg-white hover:border-gray-6 text-gray-6 cursor-pointer;
}

.wa-card-nextGen-tooltip {
  @apply border-dashed border-2 border-skyblue-5 cursor-pointer bg-gray-2 rounded-md;
}

.wa-medium-card {
  @apply p-0;
}

.wa-small-card {
  @apply p-0;
}

.focused-card {
  min-width: 216px;
  min-height: 64px;
}

.card-option {
  animation: fadeInAnimation ease 0.5s;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  @apply w-52 flex mt-2 bg-white rounded-md shadow-card items-center py-2 px-3 cursor-pointer hover:shadow-inner;
}

@keyframes fadeInAnimation {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

.card-option-title {
  @apply ml-4 whitespace-nowrap overflow-hidden overflow-ellipsis;
}

/* Card Tooltip */
.maskClassName {
  background: #fff;
  opacity: 0;
}

.highlightedMaskClassName {
  background: #fff;
}

.arrow-down {
  width: 0;
  height: 0;
  border-left: 16px solid transparent;
  border-right: 16px solid transparent;
  border-top: 16px solid #fff;
  position: absolute;
  left: 144px;
  bottom: -15px;
}

.reactour__helper {
  @apply absolute !important w-80 !important -top-36 !important -left-32 md:left-48 lg:left-64 2xl:-left-64 p-4;
}

.focus-node-width {
  width: max-content;
}

.card-menu {
  border: none !important;
  box-shadow: none !important;
  box-sizing: unset !important;
  padding: 0 !important;
  background: transparent !important;
}

#simple-popover {
  background: rgba(0, 0, 0, 0.4);
}
#simple-popoverevent {
  background: rgba(0, 0, 0, 0.8);
}

/* File Selector */
#getFile {
  display: none;
}

#getFileLabel {
  display: block;
  background-image: url("./defaultImage.png");
  width: 48px;
  height: 48px;
  cursor: pointer;
}

.card-content-img {
  min-width: 48px;
}

.cardImage {
  @apply max-w-none;
}
.focus-birth-date-truncate {
  @apply truncate;
}

.small-name-truncate{
  @apply truncate
} 

.medium-name-truncate{
  @apply truncate
}

.focus-name-truncate{
  @apply truncate
} 

/* Prevent default tooltip of Safari Browser */
@media not all and (min-resolution:.001dpcm)
{ @supports (-webkit-appearance:none) {
    .focus-name-truncate::after,
    .medium-name-truncate::after,
    .small-name-truncate::after,
    .focus-birth-date::after{
      content: " ";
      display: block;
    }
  }
}