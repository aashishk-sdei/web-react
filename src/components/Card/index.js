import React, { useEffect, useState, useRef } from "react";
import PropTypes from "prop-types";
import "./index.css";
import { titleCase } from "../utils/titlecase"
// Local Components
import GetPopOver from "./GetPopOver";
import { CardContent } from "./CardContent";
import FileSelector from "./FileSelector";
import defaultImage from "./defaultImage.png";
import { validateForm, emptyErrors } from "./validations";
import { handlePanActions, getPanPopOver, setPanPopOver } from "./helpers";

// Utils
import { cardType } from "../utils";
import { showBirthandDeath } from "shared-logics";

import MaleImg from "../../assets/images/maleVectoriconlg.svg";
import FemaleImg from "../../assets/images/femaleVectoriconlg.svg";
import OtherImg from "../../assets/images/otherVectoriconlg.svg";
import { MALE, FEMALE, getCustomImageUrl } from "../../utils";

const { FOCUSED_LIVING, FOCUSED_DECEASED, MEDIUM_IMG, MEDIUM_NEXT_GEN, SMALL_IMG, SMALL_NEXT_GEN } = cardType;
let ua = navigator.vendor.toLowerCase();
const Card = ({
    id,
    parentId,
    cFilialId,
    type,
    path,
    title,
    firstName,
    firstNameWithInitials,
    lastName,
    isLiving,
    gender,
    birth,
    birthPlace,
    death,
    deathPlace,
    imgsrc,
    relatedParentIds,
    ...props
}) => {
    const node = useRef();
    const formRef = useRef();
    const defaultClick = useRef();
    const [myObj] = useState({ id, parentId, cFilialId, type, path, title, firstName, firstNameWithInitials, lastName, isLiving, gender, birth: birth ?.RawDate, birthPlace, death: death ?.RawDate, deathPlace, imgsrc, relatedParentIds });
    const [tooltip, setTooltip] = useState(type === MEDIUM_NEXT_GEN && path === "0" && localStorage.getItem("card-tooltip") === "true" ? true : false);
    const [formData, setFormData] = useState({ id, parentId, cFilialId, type, path, title, firstName, firstNameWithInitials, lastName, isLiving, gender, birth, birthPlace, death, deathPlace, imgsrc, relatedParentIds });
    const [anchorEl, setAnchorEl] = useState(null);
    const [open, setOpen] = useState(false);
    const [popId, setPopId] = useState(undefined);
    const [subMenu, setSubMenu] = useState(false);
    const [errorMessage, setErrorMessage] = useState({});
    const checkImageURL = (url) => {
        if (url.match(/\.(jpeg|jpg|gif|png|jfif)$/)) {
            if(type === FOCUSED_LIVING || type === FOCUSED_DECEASED) return getCustomImageUrl('q=100,w=92,h=96', url);
            else if(type === MEDIUM_IMG) return getCustomImageUrl('q=100,w=88,h=88', url);
            else if(type === SMALL_IMG) return getCustomImageUrl('q=100,w=40,h=40', url);
            else return url;
        }
        else if (type === FOCUSED_LIVING || type === FOCUSED_DECEASED) {
            if (props.isOwner) return defaultImage;
            else if (!props.isOwner) return OtherImg;
            else if (gender === MALE) return MaleImg;
            else if (gender === FEMALE) return FemaleImg;
            else return OtherImg;
        }
        else return null;
    };
    const imageURL = checkImageURL(imgsrc);
    useEffect(()=>{
        setTimeout(()=>{
            props.defaultFocus(defaultClick.current, setSubMenu, setAnchorEl, setOpen, setPopId)
        }, 0)
    }, [])
    const getNextGen = () => {
        switch (type) {
            case MEDIUM_NEXT_GEN:
            case SMALL_NEXT_GEN: return true
            default: return false;
        }
    }
    const nextGen = getNextGen()

    const checkHideAddParent = (checkParentIds) => {
        const lengthOfKeys = checkParentIds ? Object.keys(checkParentIds).length : 0
        return lengthOfKeys === 2 ? true : false;
    }
    const hideAddParent = checkHideAddParent(relatedParentIds);

    const checkHideAddSibling = (checkParentIds) => {
        const lengthOfKeys = checkParentIds ? Object.keys(checkParentIds).length : 0
        return lengthOfKeys > 0 ? true : false;
    }
    const hideAddSibling = checkHideAddSibling(relatedParentIds);

    useEffect(() => {
        document.addEventListener("click", handleOutSide);

        return () => {
            document.removeEventListener("click", handleOutSide);
        };
    }, []);

    const handleOutSide = e => {
        if (node && node.current && node.current.contains(e.target)) {
            // Detect Inside
            return;
        }
        // Detect Outside
    };

    const handlePopOpen = (event) => {
        const openPop = event.currentTarget;
        setAnchorEl(openPop);
        setOpen(Boolean(openPop));
        setPopId("simple-popover");
    }

    const handlePopClose = () => {
        setAnchorEl(null);
        setOpen(false);
        setPopId(undefined);
        setSubMenu(false);
        setErrorMessage({});
        emptyErrors();
    }

    const handleOpen = (event) => {
        const clickedElementType = event.target.nodeName;
        if ((clickedElementType !== 'LABEL') && (clickedElementType !== 'INPUT')) {
            handlePanActions(event, props.handleViewBox, type);
            if (tooltip) handleCloseTour();
            else {
                handlePopOpen(event);
                setFormData({ id, parentId, cFilialId, type, path, title, firstName, firstNameWithInitials, lastName, isLiving, gender, birth, birthPlace, death, deathPlace, imgsrc, relatedParentIds });
            }
        }
    };
    //New approach to add father or mother via nodes
    const handleNextGeneration = () => {
        if (props.isOwner) props.handleNextGeneration(myObj);
    }

    const resetPanning = () => {
        if (getPanPopOver()) {
            const popOver = document.getElementById("simple-popover");
            const nodeForm = document.getElementById("nodeForm");
            popOver.style.inset = "0px 0px 0px 0px";
            nodeForm.style.marginLeft = "0px";
            setPanPopOver(false);
        }
    }
    //Old approach to add father or mother
    const handleClose = async () => {
        if (formRef.current && formRef.current.isValid) {
            if (Object.keys(errorMessage).length > 0 || Object.keys(formRef.current.errors).length > 0) {
                handlePopClose();
            } else {
                if (formRef.current.values.firstName) {
                    const result = await props.handleSubmit(formRef.current.values);
                    if (result && result.status === 200) {
                        props.handleSaveParents();
                        handlePopClose();
                    } else {
                        handleErrors(formRef, result);
                    }
                } else {
                    handlePopClose();
                }
            }
        } else {
            handlePopClose();
        }
    };

    const handleChange = (e, setFieldValue) => {
        resetPanning();
        if (e.preventDefault) e.preventDefault();
        setFieldValue(e.target.name, e.target.value);
        const errors = validateForm(e.target.name, e.target.value);
        if (errors) handleErrors(formRef, errors);
        else handleErrors(formRef, null);
    }

    const handleErrors = (myFormRef, errors) => {
        if (errors) {
            myFormRef && myFormRef.current && myFormRef.current.setErrors(errors);
            setErrorMessage(errors);
        } else {
            myFormRef && myFormRef.current && myFormRef.current.setErrors(null);
            setErrorMessage({});
        }
    }

    const handleIsLiving = (value) => {
        resetPanning();
        if (value) {
            formRef.current.setFieldValue("isLiving", true);
            formRef.current.setFieldValue("death", "");
            formRef.current.setFieldValue("deathPlace", "");
        } else formRef.current.setFieldValue("isLiving", false);
        handleErrors(formRef, null);
    }

    const handleKeyDown = (e) => {
        switch (e.keyCode) {
            // Enter
            case 13:
                handleClose();
                break;

            // Tab
            case 27:
                handlePopClose();
                break;

            default:
                break;
        }
    }

    const handleMenu = (menu) => {
        if (menu === 3) {
            resetPanning();
            setSubMenu(true);
        } else {
            handlePopClose();
            props.handleCardMenu(menu, myObj);
        }
    }

    const handleUploader = (event) => {
        handlePopClose();
        let fileList = event.target.files;
        if (imageURL.includes("data:image/png;base64")) props.handleImageUploader(fileList, myObj);
    }

    const handleCloseTour = () => {
        setTooltip(false);
        localStorage.setItem("card-tooltip", false);
        localStorage.setItem("new-tree", false);
    }
    const isOwnerTree = props.isOwner
    const getPropsForCardContent = () => {
        return { type, firstName, firstNameWithInitials, lastName, birth, birthPlace, death, deathPlace, imageURL, checkFileSelector, handleUploader,isOwnerTree, isLiving }
    }

    const getPropsForPopOver = (focusMenu) => {
        return {
            popId,
            open,
            anchorEl,
            handleClose,
            formData,
            handleChange,
            handleKeyDown,
            nextGen,
            type,
            handlePanActions,
            firstName,
            firstNameWithInitials,
            lastName,
            handleIsLiving,
            handleMenu,
            focusMenu,
            handleViewBox: props.handleViewBox,
            subMenu,
            hideAddParent,
            hideAddSibling,
            formRef,
            errorMessage,
            isOwner: props.isOwner,
            isLiving
        }
    }

    const checkFileSelector = () => {
        return props.isOwner && !imgsrc && imageURL.includes("data:image/png;base64");
    }

    
    switch (type) {
        case FOCUSED_LIVING:
            return (
                <div ref={node} className={isOwnerTree ? "wa-focused-card focused-card text-white focus-pointer":"wa-focused-card focused-card text-white"}>
                    <div className="flex living" ref={defaultClick} aria-describedby={popId} {...((isOwnerTree || (!isOwnerTree && !isLiving)) && {onClick:(handleOpen)})}>
                        <table className="w-full">
                            <tbody>
                                <tr>
                                    <td className="align-top w-12">
                                      {(isOwnerTree || (!isOwnerTree && !isLiving)) ? (checkFileSelector() ? <FileSelector handleUploader={handleUploader} /> : <img src={imageURL} alt="card-pic-medium" className="w-full h-12 rounded-lg" />):<img src={OtherImg} alt="card-pic-medium" className="w-full h-12 rounded-lg" />}
                                    </td>
                                    <td>
                                        <div className="flex focus-node-width p-0">
                                            <div className="pl-2">
                                                <div className="text-base typo-font-medium  max-w-xs mb-0.5 truncatetext focus-name-truncate">{isOwnerTree || (!isOwnerTree && !isLiving) ? `${titleCase(firstNameWithInitials)} ${titleCase(lastName)}`: "Private"}</div>
                                                <div className="text-xs my-0 ml-3.92 max-w-city truncate">{`B: ${isOwnerTree || (!isOwnerTree && !isLiving) ? titleCase(birth.RawDate) : "Living"}`}</div>
                                                <div className="text-xs my-0 ml-3.92 max-w-city truncate">{isOwnerTree || (!isOwnerTree && !isLiving) ? `${titleCase(birthPlace)}`: ''}</div>
                                            </div>
                                        </div>
                                    </td>
                                </tr>
                            </tbody>
                        </table>
                    </div>
                    <GetPopOver {...getPropsForPopOver(false)} content={<CardContent {...getPropsForCardContent()} />} />
                </div>
            );

        case FOCUSED_DECEASED:
            return (
                <div ref={node} className="wa-focused-card focused-card text-white  focus-pointer">
                    <div className="flex deceased" aria-describedby={popId} onClick={handleOpen}>
                        <table className="w-full ">
                            <tbody>
                                <tr>
                                    <td className="align-top w-12">
                                        {checkFileSelector() ? <FileSelector handleUploader={handleUploader} /> : <img src={imageURL} alt="card-pic-xlarge" className="h-12 w-12 rounded-lg" />}
                                    </td>
                                    <td className="">
                                        <div className="flex focus-node-width">
                                            <div className="pl-2">
                                                <div className="text-base typo-font-medium  max-w-xs mb-0.5 truncatetext focus-name-truncate">{`${titleCase(firstNameWithInitials)} ${titleCase(lastName)}`}</div>
                                                <div className="mb-2">
                                                    <p className="text-xs my-0 ml-3.92 max-w-city truncate">{`B: ${titleCase(birth.RawDate)}`}</p>
                                                    <p className=" text-xs my-0 ml-3.92 max-w-city truncate">{`${titleCase(birthPlace)}`}</p>
                                                </div>
                                                <div>
                                                    <p className="text-xs my-0 ml-3.92 max-w-city truncate">{`D: ${titleCase(death.RawDate)}`}</p>
                                                    <p className="text-xs my-0 ml-3.92 max-w-city truncate">{`${titleCase(deathPlace)}`}</p>
                                                </div>
                                            </div>
                                        </div>
                                    </td>
                                </tr>
                            </tbody>
                        </table>
                    </div>
                    <GetPopOver {...getPropsForPopOver(false)} content={<CardContent {...getPropsForCardContent()} />} />
                </div>
            );

        case MEDIUM_NEXT_GEN:
            return (<div ref={node} className={props.isOwner ? "wa-card-nextGen-hover" : "wa-card-nextGen"}>
                    <div className="flex" aria-describedby={popId} onClick={handleNextGeneration}>
                        <div className="px-4 py-3.5">
                            <div className="text-xs typo-font-medium truncatetext w-44 text-gray-6">+ {title}</div>
                        </div>
                    </div>
                    <GetPopOver {...getPropsForPopOver(true)} />
                </div>
            );
        case SMALL_IMG:
            return (
                <div ref={node} className="wa-card-small">
                    <div className="flex" aria-describedby={popId} onClick={handleOpen}>
                        {(isOwnerTree || (!isOwnerTree && !isLiving)) && imageURL && <span className="h-6 w-6" > <img src={imageURL} alt="card-pic-medium" className="w-5 h-5 m-0.5 rounded-l cardImage" /></span>}
                        <div className={`${imageURL ? 'px-2' : 'px-4'} py-1`}>
                            <div className={`w-44 text-xs typo-font-medium truncatetext small-name-truncate`}>{(isOwnerTree || (!isOwnerTree && !isLiving)) ? `${titleCase(firstNameWithInitials)} ${titleCase(lastName)}` : "Private"}</div>
                        </div>
                    </div>
                    <GetPopOver {...getPropsForPopOver(true)} content={<CardContent {...getPropsForCardContent()} />} />
                </div>
            );

        case SMALL_NEXT_GEN:
            return (
                <div ref={node} className={props.isOwner ? "wa-card-nextGen-hover" : "wa-card-nextGen h-6"}>
                    <div className="flex" aria-describedby={popId} onClick={handleNextGeneration}>
                        {ua !== "apple computer, inc." && (
                            <div className="px-4 py-0.5">
                                <div className="text-xs typo-font-medium  truncatetext w-44 text-gray-6">+ {title}</div>
                            </div>
                        )}
                        {ua === "apple computer, inc." && (
                            <div className="px-4">
                                <div className="text-xs typo-font-medium  truncatetext w-44 text-gray-6">+ {title}</div>
                            </div>
                        )}
                    </div>
                    <GetPopOver {...getPropsForPopOver(true)} />
                </div>
            );

        case MEDIUM_IMG:
        default:
            return (
                <div ref={node} className="wa-card-medium">
                    <div className="flex max-w-fitcontent" aria-describedby={popId} onClick={handleOpen}>
                        <span className="h-12">
                            {(isOwnerTree || (!isOwnerTree && !isLiving)) && imageURL && <img src={imageURL} alt="card-pic-medium" className="h-11 w-11 m-0.5 rounded-l cardImage" />}
                        </span>
                        <div className={`${(isOwnerTree || (!isOwnerTree && !isLiving)) && imageURL ? 'px-2' : 'px-4'} py-1 flex flex-col justify-center  max-w-fitcontent`}>
                            <div className={`${(isOwnerTree || (!isOwnerTree && !isLiving)) && imageURL ? 'w-36' : 'w-46'} text-xs typo-font-medium max-h-fitcontent max-w-fitcontent truncatetext medium-name-truncate mb-0.5`}>{(isOwnerTree || (!isOwnerTree && !isLiving)) ? `${titleCase(firstNameWithInitials)} ${titleCase(lastName)}` : "Private"}</div>
                            <p className="text-vs my-0 truncate w-36">{showBirthandDeath(birth.Year, death.Year, isLiving, isOwnerTree)}</p>
                        </div>
                    </div>
                    <GetPopOver {...getPropsForPopOver(true)} content={<CardContent {...getPropsForCardContent()} />} />
                </div>
            );
    }
}

Card.propTypes = {
    id: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number
    ]),
    type: PropTypes.oneOf([FOCUSED_LIVING, FOCUSED_DECEASED, MEDIUM_IMG, MEDIUM_NEXT_GEN, SMALL_IMG, SMALL_NEXT_GEN]),
    title: PropTypes.string,
    firstName: PropTypes.string,
    firstNameWithInitials: PropTypes.string,
    lastName: PropTypes.string,
    isLiving: PropTypes.bool,
    gender: PropTypes.string,
    birth: PropTypes.string,
    birthPlace: PropTypes.string,
    death: PropTypes.string,
    deathPlace: PropTypes.string
};

Card.defaultProps = {
    id: "id",
    type: MEDIUM_IMG,
    title: "Kathryns Mother",
    firstName: "Kimberly Marie",
    firstNameWithInitials: "Kimberly M",
    lastName: "Walker",
    isLiving: true,
    gender: "Male",
    birth: "1 Jan 1940",
    birthPlace: "Los Angeles, California",
    death: "1 Dec 1990",
    deathPlace: "Los Angeles, California"
}

export default Card;