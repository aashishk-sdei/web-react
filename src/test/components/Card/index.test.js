import React from 'react';
import { shallow } from 'enzyme';

import Card from '../../../components/Card';
import { cardType } from "../../../components/utils";

const { FOCUSED_LIVING } = cardType;

const blankId = "00000000-0000-0000-0000-000000000000";

test('should test Card component with list of props', () => {
    const wrapper = shallow(
        <Card 
            id={blankId}
            parentId={blankId}
            cFilialId={blankId}
            type={FOCUSED_LIVING}
            path={"0/0"}
            title="Add Father"
            firstName="John Walker"
            firstNameWithInitials="J W"
            lastName="Alice"
            isLiving={true}
            gender="Male"
            birth="01 Jan 1885"
            birthPlace="Utah, USA"
            death="20 Oct 1935"
            deathPlace="Utah, USA"
            imgsrc="imgsrc"
            relatedParentIds={blankId}
        />
    );
    expect(wrapper).toMatchSnapshot();
});