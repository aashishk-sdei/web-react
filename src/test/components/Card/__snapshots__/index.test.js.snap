// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Card component with list of props 1`] = `ShallowWrapper {}`;
